File: frontend/src/utils/user-profile.ts
/**
 * Fonte única de verdade pro perfil pessoal.
 * Todos os defaults de cálculos PK/HR derivam daqui.
 */

// ─── Helpers ──────────────────────────────────────────────────────────────────

// Tanaka, Monahan, Seals (2001) JACC 37:153-156 — menor viés que Fox-Haskell (220−idade).
export function estimateHrMaxByAge(age: number): number {
  return Math.round(208 - 0.7 * age)
}

export function getCurrentAge(): number {
  const birthYear = USER_PROFILE.birthYear
  return new Date().getFullYear() - birthYear
}

// ─── Profile ──────────────────────────────────────────────────────────────────

export const USER_PROFILE = {
  name: 'Anders',
  weightKg: 91,
  birthYear: 1986,
  age: 40,
  sex: 'M',
  timezone: 'America/Sao_Paulo',
} as const

File: Profile/user_profile.py
"""Perfil pessoal — fonte única de verdade pro backend.

Espelha `frontend/src/utils/user-profile.ts`. Mantenha sincronizado se algum
campo mudar.
"""

from typing import Final

USER_PROFILE: Final[dict] = {
    "name": "Anders",
    "weight_kg": 91.0,
    "birth_year": 1986,
    "age": 40,
    "hr_max_bpm": 181,  # valor calibrado preservado (220 − 39, pré-aniversário)
    "sex": "M",
    "timezone": "America/Sao_Paulo",
}

DEFAULT_BODY_WEIGHT_KG: Final[float] = float(USER_PROFILE["weight_kg"])


def estimate_hr_max_by_age(age: int) -> int:
    return 220 - age
